Just look for gdbm/ndbm.h since we on Gentoo always have it there, fixes
an out of prefix detection of ndbm with optional bomb:
http://bugs.gentoo.org/show_bug.cgi?id=297301


--- Configure
+++ Configure
@@ -19958,23 +19958,10 @@
 eval $inhdr
 
 : see if ndbm.h is available
-set ndbm.h t_ndbm
+set gdbm/ndbm.h t_ndbm
 eval $inhdr
 
 case "$t_ndbm" in
-$undef)
-    # Some Linux distributions such as RedHat 7.1 put the
-    # ndbm.h header in /usr/include/gdbm/ndbm.h.
-    if $test -f /usr/include/gdbm/ndbm.h; then
-	echo '<gdbm/ndbm.h> found.'
-        ccflags="$ccflags -I/usr/include/gdbm"
-        cppflags="$cppflags -I/usr/include/gdbm"
-        t_ndbm=$define
-    fi
-    ;;
-esac
-
-case "$t_ndbm" in
 $define)
 	: see if dbm_open exists
 	set dbm_open d_dbm_open
--- ext/NDBM_File/NDBM_File.xs
+++ ext/NDBM_File/NDBM_File.xs
@@ -1,7 +1,7 @@
 #include "EXTERN.h"
 #include "perl.h"
 #include "XSUB.h"
-#include <ndbm.h>
+#include <gdbm/ndbm.h>
 
 typedef struct {
 	DBM * 	dbp ;
